docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,503 @@
+
+```
+    // ===== Members functions that implement the DeviceAttestationCredentialsProvider
+    CHIP_ERROR GetCertificationDeclaration(MutableByteSpan & outBuffer) override;
+    CHIP_ERROR GetFirmwareInformation(MutableByteSpan & out_firmware_info_buffer) override;
+    CHIP_ERROR GetDeviceAttestationCert(MutableByteSpan & outBuffer) override;
+    CHIP_ERROR GetProductAttestationIntermediateCert(MutableByteSpan & outBuffer) override;
+    CHIP_ERROR SignWithDeviceAttestationKey(const ByteSpan & digestToSign, MutableByteSpan & outSignBuffer) override;
+
+    // ===== Members functions that implement the CommissionableDataProvider
+    CHIP_ERROR GetSetupDiscriminator(uint16_t & setupDiscriminator) override;
+    CHIP_ERROR SetSetupDiscriminator(uint16_t setupDiscriminator) override;
+    CHIP_ERROR GetSpake2pIterationCount(uint32_t & iterationCount) override;
+    CHIP_ERROR GetSpake2pSalt(MutableByteSpan & saltBuf) override;
+    CHIP_ERROR GetSpake2pVerifier(MutableByteSpan & verifierBuf, size_t & verifierLen) override;
+    CHIP_ERROR GetSetupPasscode(uint32_t & setupPasscode) override;
+    CHIP_ERROR SetSetupPasscode(uint32_t setupPasscode) override;
+
+    // ===== Members functions that implement the DeviceInstanceInfoProvider
+    CHIP_ERROR GetVendorName(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetVendorId(uint16_t & vendorId) override;
+    CHIP_ERROR GetProductName(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetProductId(uint16_t & productId) override;
+    CHIP_ERROR GetSerialNumber(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetManufacturingDate(uint16_t & year, uint8_t & month, uint8_t & day) override;
+    CHIP_ERROR GetHardwareVersion(uint16_t & hardwareVersion) override;
+    CHIP_ERROR GetHardwareVersionString(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetRotatingDeviceIdUniqueId(MutableByteSpan & uniqueIdSpan) override;
+```
+
+You can read the factory data set from the device's Flash memory in different ways depending on purpose and format. In the nRF Connect example, the factory data is stored in CBOR format. Then a device uses the [Factory Data Parser](../../src/platform/nrfconnect/FactoryDataParser.h) to read out raw data, decode them, and store them in the `FactoryData` structure. The [Factor Data Provider](../../src/platform/nrfconnect/FactoryDataProvider.c) implementation uses this parser to get all needed factory data parameters and provide them to the Matter core.
+
+The last thing to do is add a newly created parser and accessor to the build system by replacing the default ones. To do that, we recommend putting new implementations into *connectedhomeip-root/src/platform/nrfconnect* directory, editing the [BUILD.gn](../../src/platform/nrfconnect/BUILD.gn), finding out section:
```

Comment:
I think a user shouldn't be required to modify the core source files. They can just add their parser/accessor to their project. Also, Kamil has added FACTORY_DATA_CUSTOM_BACKEND (IIRC) that can be used to disable building both the default and our implementations, so we can recommend setting it.

Reply:
Ok, I missed the FACTORY_DATA_CUSTOM_BACKEND config meaning. Now it seems to be easier to explain it. :)